src/app/encoder.cpp

```diff
@@ -2974,21 +2962,15 @@ PacketBufferHandle encodeMediaPlaybackClusterReadClusterRevisionAttribute(uint8_
  * Command AddThreadNetwork
  */
 PacketBufferHandle encodeNetworkCommissioningClusterAddThreadNetworkCommand(uint8_t seqNum, EndpointId destinationEndpoint,
-                                                                            char * operationalDataset, uint64_t breadcrumb,
+                                                                            chip::BytesData operationalDataset, uint64_t breadcrumb,
                                                                             uint32_t timeoutMs)
 {
     COMMAND_HEADER("AddThreadNetwork", NETWORK_COMMISSIONING_CLUSTER_ID);
-    size_t operationalDatasetStrLen = strlen(operationalDataset);
-    if (!CanCastTo<uint8_t>(operationalDatasetStrLen))
-    {
-        ChipLogError(Zcl, "Error encoding %s command. String too long: %d", kName, operationalDatasetStrLen);
-        return PacketBufferHandle();
-    }
     buf.Put8(kFrameControlClusterSpecificCommand)
         .Put8(seqNum)
         .Put8(ZCL_ADD_THREAD_NETWORK_COMMAND_ID)
-        .Put(static_cast<uint8_t>(operationalDatasetStrLen))
-        .Put(operationalDataset)
+        .Put(static_cast<uint8_t>(operationalDataset.Length()))
```

Comment:
Why is this cast safe?

I'm going to stop asking this question, but this problem is _everywhere_ in this file.